File: 6-prefix-sum/1-intro/main.py
# Prefix Sum - Python Example

def prefix_sum(arr):
    prefix = [0] * len(arr)
    prefix = arr
    for i in range(1, len(arr)):
        prefix[i] = prefix[i-1] + arr[i]
    return prefix

arr = [1, 2, 3, 4]
prefix = prefix_sum(arr)
print("Prefix sum array:", prefix)
# Sum of range 1 to 3:
result = prefix[3] - prefix
print("Sum from index 1 to 3:", result)

File: 6-prefix-sum/1-intro/main.java
// Prefix Sum - Java Example

class PrefixSum {

    public static int[] prefixSum(int[] arr) {
        int[] prefix = new int[arr.length];
        prefix[0] = arr[0];
        for (int i = 1; i < arr.length; i++) {
            prefix[i] = prefix[i - 1] + arr[i];
        }
        return prefix;
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4};
        int[] prefix = prefixSum(arr);

        System.out.print("Prefix sum array: ");
        for (int val : prefix) {
            System.out.print(val + " ");
        }
        System.out.println();

        int result = prefix[3] - prefix[0];
        System.out.println("Sum from index 1 to 3: " + result);
    }
}
